predicators/structs.py

```diff
@@ -415,6 +415,42 @@ def goal_holds(self, state: State) -> bool:
 DefaultTask = Task(DefaultState, set())
 
 
+@dataclass(frozen=True, eq=False)
+class EnvironmentTask:
+    """An initial observation and goal description.
+
+    Environments produce environment tasks and agents produce and solve tasks.
+
+    In fully observed settings, the init_obs will be a State and the
+    goal_description will be a Set[GroundAtom]. For convenience, we can convert
+    an EnvironmentTask into a Task in those cases.
+    """
+    init_obs: Observation
+    goal_description: GoalDescription
```

Comment:
right, the `done` bit idea is still important, but I implemented that in the cogman PR instead of here. specifically, I added a `goal_reached()` method in `BaseEnv`: https://github.com/Learning-and-Intelligent-Systems/predicators/pull/1431/files#diff-6bb438d682fedf907f3cde635a6431633c80ae51fca91ab4b55c920bf912b6abR190

Reply:
this makes sense for now!